code/modules/telesci/rcs.dm

```diff
@@ -38,84 +48,79 @@
 	QDEL_NULL(rcell)
 	return ..()
 
+/**
+  * Used to select telepad location.
+  */
 /obj/item/rcs/attack_self(mob/user)
-	if(!emagged)
+	if(teleporting)
+		to_chat(user, "<span class='warning'>Error: Unable to change destination while in use.</span>")
 		return
-	if(mode == RCS_MODE_CALIBRATED)
-		mode = RCS_MODE_UNCALIBRATED
-		playsound(get_turf(src), 'sound/effects/pop.ogg', 50, 0)
-		to_chat(user, "<span class='caution'>The telepad locator has become uncalibrated.</span>")
-	else
-		mode = RCS_MODE_CALIBRATED
-		playsound(get_turf(src), 'sound/effects/pop.ogg', 50, 0)
-		to_chat(user, "<span class='caution'>You calibrate the telepad locator.</span>")
-
-/obj/item/rcs/emag_act(user as mob)
+
+	var/list/L = list() // List of avaliable telepads
+	var/list/areaindex = list() // Telepad area location
+	for(var/obj/machinery/telepad_cargo/R in GLOB.machines)
+		if(R.stage)
+			continue
+		var/turf/T = get_turf(R)
+		var/locname = T.loc.name // The name of the turf area. (e.g. Cargo Bay, Experimentation Lab)
+
+		if(areaindex[locname]) // If there's another telepad with the same area, increment the value so as to not override (e.g. Cargo Bay 2)
+			locname = "[locname] ([++areaindex[locname]])"
+		else // Else, 1
+			areaindex[locname] = 1
+		L[locname] = R
+
+	if(emagged) // Add an 'Unknown' entry at the end if it's emagged
+		L += "**Unknown**"
+
+	var/select = input("Please select a telepad.", "RCS") in L
+	if(select == "**Unknown**") // Randomise the teleport location
+		rand_x = rand(50, 200)
+		rand_y = rand(50, 200)
+		pad = locate(rand_x, rand_y, 6)
```

Comment:
There's a problem here where this tries to set `pad` to something at the given location, but there isn't actually a pad there, so `pad` ends up being null. Which means when you try to teleport something, you get the "Error: No telepad selected." message.

Also why the hardcoded Z-6 here?

Reply:
I've not been able to reproduce this, can you elaborate?

As for the z6, presumably so that it's always in space? I could make it z1 with a small chance of being 6, or just make it always on the station level.

Reply:
In the location selection menu, when I choose **Unknown**, and then click on a crate, it says "Error: No telepad selected."